packages/ddl-shim/src/upgrade.ts

```diff
@@ -722,7 +842,26 @@ class DDLUpgrade {
     }
 }
 
-export function upgrade(ddl: DDL1.IDDL, baseUrl?: string, wuid?: string, toLowerCase: boolean = true): DDL2.Schema {
-    const ddlUp = new DDLUpgrade(ddl, baseUrl, wuid, toLowerCase);
-    return ddlUp.write();
+export function upgrade(ddl: DDL1.IDDL, baseUrl?: string, wuid?: string, toLowerCase: boolean = true, dermatologyJson: any = {}): DDL2.Schema {
+    const ddlUp = new DDLUpgrade(ddl as DDL1.IDDL, baseUrl, wuid, toLowerCase, dermatologyJson);
+    const ret = ddlUp.write();
+    console.log("ret", ret);
+    return ret;
+}
+
+function findKeyVal(object: any, key: any, val: any) {
+    let value;
+
+    Object.keys(object).some(k => {
+        if (k === key && object[k] === val) {
+            value = object;
+            return true;
+        }
+        if (object[k] && typeof object[k] === "object") {
+            value = findKeyVal(object[k], key, val);
+            return value !== undefined;
+        }
+        return false;
+    });
```

Comment:
This looks very inefficient

Reply:
It's actually really great to be able to recursively search a dermatology object for a widget's `__id`. It allows us to grab the dermatology widget properties that we're looking for without needing to hard code conditions for specific versions etc.

Reply:
PropertyExt already has a "widgetWalker" and a "propertyWalker" which you could use (or a `d3Select("#id").datum()` if the widget is rendered).